### DISCUSSION TITLE
Upgrade TypeScript to v5.x to support modern libraries like Fontawesome

### DISCUSSION THREAD

**Discussion by chrissnyder2337** [Sitecore/jss#1852] [Ideas]
### Is your suggestion related to a problem? Please describe.

It needs to be able to support modern dependencies which leverage a different bundler.

Upgrade to typescript 5 to allow support of libraries like Fontawesome 6 which needs the `moduleResolution` set to `bundler`, only available in 5.0.2.

See https://docs.fontawesome.com/web/dig-deeper/kit-package-api#typescriptIt needs to be able to support modern dependencies which leverage a different bundler.

### Describe the solution you'd like

Upgrade to typescript 5 to allow support of libraries like Fontawesome 6 which needs the `moduleResolution` set to `bundler`, only available in 5.0.2.

**Comment by mg-aceik** [comment]
I updated typescript myself since 21.4+ which works nicely. But agreed.

**Comment by yavorsk** [comment]
Hi @chrissnyder2337 , thanks for submitting this. I have created an item in our backlog re typescript upgrade :)

**Comment by illiakovalenko** [comment]
I'm revisiting older discussions. Since our solution is currently using TypeScript 5.4, I'll mark this discussion as closed and resolved
